presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -255,7 +256,7 @@ public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
     {
         consumer.accept(dictionary, dictionary.getRetainedSizeInBytes());
         consumer.accept(ids, sizeOf(ids));
-        consumer.accept(this, (long) INSTANCE_SIZE);
+        consumer.accept(this, (long) INSTANCE_SIZE + DICTIONARY_ID_SIZE);
```

Comment:
I think it's better to stick to the specification/javadoc. Please check with @highker if you want to understand better what the spec really expects from the implementation.